monai/transforms/io/array.py

```
@@ -210,7 +210,7 @@ def __init__(
                 try:
                     self.register(the_reader(*args, **kwargs))
                 except OptionalImportError:
-                    warnings.warn(
+                    raise ImportError(
                         f"required package for reader {_r} is not installed, or the version doesn't match requirement."
                     )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)